plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepository.java

```diff
@@ -60,31 +60,23 @@
  */
 public class AzureRepository extends BlobStoreRepository {
 
-    public final static String TYPE = "azure";
-
-    static public final class Defaults {
-        public static final String CONTAINER = "elasticsearch-snapshots";
-        public static final ByteSizeValue CHUNK_SIZE = new ByteSizeValue(64, ByteSizeUnit.MB);
-        public static final Boolean COMPRESS = false;
-    }
+    private static final ByteSizeValue MAX_CHUNK_SIZE = new ByteSizeValue(64, ByteSizeUnit.MB);
 
+    public final static String TYPE = "azure";
 
-    static public final class Repository {
-        public static final String ACCOUNT = "account";
-        public static final String LOCATION_MODE = "location_mode";
-        public static final String CONTAINER = "container";
-        public static final String CHUNK_SIZE = "chunk_size";
-        public static final String COMPRESS = "compress";
-        public static final String BASE_PATH = "base_path";
+    public static final class Repository {
+        public static final Setting<String> ACCOUNT_SETTING = Setting.simpleString("account", false, Setting.Scope.CLUSTER);
+        public static final Setting<String> CONTAINER_SETTING = new Setting<>("container", "elasticsearch-snapshots", Function.identity(), false, Setting.Scope.CLUSTER);
+        public static final Setting<String> BASE_PATH_SETTING = Setting.simpleString("base_path", false, Setting.Scope.CLUSTER);
+        public static final Setting<String> LOCATION_MODE_SETTING = Setting.simpleString("location_mode", false, Setting.Scope.CLUSTER);
+        public static final Setting<ByteSizeValue> CHUNK_SIZE_SETTING = Setting.byteSizeSetting("chunk_size", new ByteSizeValue(64, ByteSizeUnit.MB), false, Setting.Scope.CLUSTER);
```

Comment:
Ah, I thought they were unrelated. That's why I kept them separate. I can change that. Thanks for clarifying.